### PR TITLE
React Calculator App - Milestone 3

### PR DESCRIPTION
**Pull request by 300ms**
- Add a little bit of styling to the project

### CODE REVIEW

File: src/index.css [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #root {
   width: 100%;
   height: 100%;
+  margin: 10px;
```
Comment (whiz25) [inline src/index.css:11]: - `.display-container` should have a height of `100px`, background of `gray`, aligned to the right, and presented in white and bold text.
- the calculator should have a width of 700px and the display mode should be `flex`
- the ButtonPanel component should have a row with height of 100px
- the Button component should be 25% of the container it's rendered in except for the 0 button which should be 50% of the it's parent container, each button should have centered text, black in colour, default color should be orange, each button should have color and wide as incoming props. If the given button is wide, it should take up 50% of the space else 25%. You need to perform a check like `wide === true ? 50% : 25%`
- each Button should have borders

**Review by whiz25 — CHANGES_REQUESTED** [review]
## changes required 🙏🏿 🙏🏿 🙏🏿 
Nice work on the styling. You came a long way.
### changes 🔕 🔕 🔕 
- your app should look exactly as [this](https://github.com/microverseinc/project-react-calculator/blob/master/images/calculator.png)
- check comments for more suggestions

### Martin Afani
#### Slack: whiz25

File: src/index.css (hunk 1 above)
Comment (300ms) [reply src/index.css:11]: Implemented all of them. Thanks for the suggestions :)

File: src/components/Button.js [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
 Button.propTypes = {
   value: PropTypes.string,
+  bgColor: PropTypes.string,
+  width: PropTypes.number,
 };
 
 Button.defaultProps = {
   value: '',
+  bgColor: '#e0e0e0',
+  width: 25,
```
Comment (rindrajosia) [inline src/components/Button.js]: - [x] [REQUIRED] width is `boolean`, true means that button take 50% of space (used for 0 button) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/WideProps.png))

File: src/components/Button.js [hunk 3]
```diff
@@ -1,16 +1,25 @@
 import PropTypes from 'prop-types';
 
 const Button = (props) => {
-  const { value } = props;
-  return <button className="button" type="button">{value}</button>;
+  const { value, bgColor, width } = props;
+
+  return (
+    <button className="button" type="button" style={{ backgroundColor: bgColor, width: `${width}%` }}>
```
Comment (rindrajosia) [inline src/components/Button.js]: - [x] [REQUIRED] Please, don't use inline styling as it is a bad practice.

File: src/index.css [hunk 4]
```diff
@@ -1,20 +1,31 @@
...
 #root {
-  width: 100%;
+  width: 700px;
   height: 100%;
+  margin: 10px;
+  display: flex;
+  flex-direction: column;
 }
 
 .display-container {
```
Comment (rindrajosia) [inline src/index.css:16]: - [x] [REQUIRED] result should be aligned to the right ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/display-alignment.png))
- [x] [REQUIRED] result should be presented with white and bold text ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/display-text.png))

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------

File: src/components/Button.js (hunk 2 above)
Comment (300ms) [reply src/components/Button.js]: 'width' prop converted to boolean value

File: src/components/Button.js (hunk 3 above)
Comment (300ms) [reply src/components/Button.js]: Inline styling is removed, using classNames instead.

File: src/index.css (hunk 4 above)
Comment (300ms) [reply src/index.css:16]: Result has been aligned to the right and presented with white and bold text. Also, font sizes has been changed.

File: .history/src/components/Button_20210901063032.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+Button.propTypes = {
+  value: PropTypes.string,
+  bgColor: PropTypes.string,
+  width: PropTypes.number,
+};
+
+Button.defaultProps = {
+  value: '',
+  bgColor: '',
```
Comment (rindrajosia) [inline .history/src/components/Button_20210901063032.js:34]: - [ ] [ REQUIRED] default `color` for `button` is `orange` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/colorProps.png))

File: src/components/Button.js [hunk 6]
```diff
@@ -1,16 +1,38 @@
...
 
 Button.propTypes = {
   value: PropTypes.string,
+  bgColor: PropTypes.string,
+  width: PropTypes.number,
 };
 
 Button.defaultProps = {
   value: '',
+  bgColor: '',
```
Comment (rindrajosia) [inline src/components/Button.js]: - [x] [ REQUIRED] default `color` for `button` is `orange` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/colorProps.png)). Here, the default value of `bgColor` is `an empty string`

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️

- [x] I got an error related to propType when I open your APP on my local:



_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/components/Button.js (hunk 6 above)
Comment (300ms) [reply src/components/Button.js]: Default color is light-gray and it is implemented now.

Comment (300ms) [reply src/components/Button.js]: I've fixed the "propType" error too.

Comment (rindrajosia) [reply src/components/Button.js]: > Default color is light-gray and it is implemented now.

The default color should be orange

**Review by whiz25 — APPROVED** [review]
## approved 👍🏿👍🏿👍🏿
Nice work making the requested changes. Merge and proceed to the next milestone.

### Martin Afani
#### Slack whiz25

**Pull request merged**